activation/activation.go

```diff
@@ -205,65 +197,60 @@ func NewBuilder(conf Config, nodeID types.NodeID, signer signer, db atxDBProvide
 		syncer:            syncer,
 		log:               log,
 		poetRetryInterval: defaultPoetRetryInterval,
-		status:            smeshingStatusIdle,
 	}
 	for _, opt := range opts {
 		opt(b)
 	}
-	b.runCtx, b.stop = context.WithCancel(b.runCtx)
 	return b
 }
 
 // Smeshing returns true iff atx builder started.
 func (b *Builder) Smeshing() bool {
-	b.mtx.Lock()
-	defer b.mtx.Unlock()
-
-	return b.status == smeshingStatusStarted
+	return b.started.Load()
```

Comment:
could this be `return b.exit != nil`

Reply:
was hard to recall thy i did it :)

StopSmeshing wasn't blocking previously, so i wanted to guarantee that IsSmeshing doesn't get stuck in a weird way if it will be called concurrently with StopSmeshing.

other reason was to set it to True only after StartSession was successful

Reply:
so it's possible that `Smeshing()` returns false, client thought it's not starting and call `StartSmeshing()` again?
why not just hold the lock until after `StartSession` success? is that an expensive call?

Reply:
i think this is when post file is created. in general i agree that it seems note very clear from ux, maybe it makes sense to change it so that smeshing is true even before file is created and false only on error or if stopped.

Reply:
i agree. i think it make more sense for smeshing to be true before the post file is created, and set it back to false if post file failed to create. when the post file fails to create, it should behave like StopSmeshing() has been called.